esmvalcore/config/_config_object.py

```diff
@@ -92,7 +92,7 @@ def __init__(self, *args, **kwargs):
             "Do not instantiate `Config` objects directly, this will lead "
             "to unexpected behavior. Use `esmvalcore.config.CFG` instead."
         )
-        warnings.warn(msg, UserWarning)
+        warnings.warn(msg, UserWarning, stacklevel=1)
```

Comment:
```suggestion
        warnings.warn(msg, UserWarning, stacklevel=2)
```

so that it warns about the piece of code that calls `Config()`?